packages/rsocket-core/src/RequestChannelStream.ts

```diff
@@ -276,7 +277,7 @@ export class RequestChannelRequesterStream
     }
 
     this.stream.send({
-      type: this.inboundDone ? FrameTypes.ERROR : FrameTypes.CANCEL,
+      type: inboundDone ? FrameTypes.ERROR : FrameTypes.CANCEL,
```

Comment:
Pretty sure this needs to be `inboundDone` rather than `this.inboundDone` as we change the value to `this.inboundDone` just before doing this comparison.